cassiopeia/preprocess/lineageGroup_utils.pyx

```diff
@@ -11,12 +18,38 @@ from scipy import cluster
 #import rpy2.robjects as robjects
 #from rpy2.robjects.packages import importr
 from tqdm import tqdm
+from typing import Dict, List, Tuple
 #from rpy2.robjects import r, numpy2ri
 import time
 import yaml
```

Comment:
Convention for imports: alphabetical, but for sys and os import at the top with `os` coming first. Leave an empty line between those system-level imports and the rest of your packages.